File: src/N83/ToDo/ToDo.ClientApp/src/modules/todos/models/ToDoItem.ts
import type { Guid } from "guid-typescript";
export class ToDoItem {
    id!: Guid;
    title!: string;
    isDone!: boolean;
    isFavorite!: boolean;
    dueTime!: Date;
    reminderTime!: Date;
    constructor() {
        this.isDone = false;
        this.isFavorite = false;
        const now = new Date();
        this.dueTime = new Date(now.getTime());
        this.dueTime.setHours(now.getHours() + 3);
        this.reminderTime = new Date(now.getTime());
        this.reminderTime.setHours(now.getHours() + 1);
    }
}

File: src/N83/ToDo/ToDo.ServerApp/ToDo.Domain/Entities/TodoItem.cs
using ToDo.Domain.Common.Entities;

namespace ToDo.Domain.Entities;

public class TodoItem : AuditableEntity
{
    public string Title { get; set; } = default!;

    public bool IsDone { get; set; }

    public bool IsFavorite { get; set; }

    public DateTimeOffset DueTime { get; set; }

    public DateTimeOffset ReminderTime { get; set; }
}